packages/sdk/src/send/send-get-events.js

```diff
@@ -4,31 +4,42 @@ import {unary as defaultUnary} from "./unary"
 
 const u8ToHex = u8 => Buffer.from(u8).toString("hex")
 const hexBuffer = hex => Buffer.from(hex, "hex")
+const isHeightRangeRequest = ix => ix.events.start !== null
+const isBlockIdsRequest = ix => !!ix.events.blockIds
 
-export async function sendGetEvents(ix, opts = {}) {
+async function getEventsForHeightRange(eventType, start, end, opts, tag) {
   const unary = opts.unary || defaultUnary
-  
-  ix = await ix
- 
-  let res
-  const req = ix.events.start ? new GetEventsForHeightRangeRequest() : new GetEventsForBlockIDsRequest()
-  req.setType(ix.events.eventType)
-  
-  if (ix.events.start) {
-    req.setStartHeight(Number(ix.events.start))
-    req.setEndHeight(Number(ix.events.end))
-
-    res = await unary(opts.node, AccessAPI.GetEventsForHeightRange, req)
-  } else {
-    ix.events.blockIds.forEach(id =>
-      req.addBlockIds(hexBuffer(id))
-    )
 
-    res = await unary(opts.node, AccessAPI.GetEventsForBlockIDs, req)
-  }
+  const req = new GetEventsForHeightRangeRequest()
+
+  req.setType(eventType)
+  req.setStartHeight(start)
+  req.setEndHeight(end)
+
+  const res = unary(opts.node, AccessAPI.GetEventsForHeightRange, req)
+
+  return formatResponse(res, tag)
+}
+
+async function getEventsForBlockIDs(eventType, blockIds, opts, tag) {
```

Comment:
I don't really agree here.

The function was doing two things before. Now that function decides which of two things to do and then passes responsibility to a specialized function to deal with the choice. If the two specialized functions need to share logic you can pull that out into smaller functions, or not.

> GetEventsForHeightRange *and* GetEventsForBlockIDs

Seeing the word *and* in a description for what a function deals with is usually a bad sign.

Reply:
Yeah, to be honest I don't see how this introduces a new code pattern. 

The current function is exhibiting two different behaviours, determined by the fields set inside a parameter that is 3 levels deep (`ix.events.start`). That's really opaque behaviour and was a bit hard to debug.

The logic already branches -- why not make it explicit?

This also handles the implicit third case in which neither a block range nor IDs is passed.

IMO the `sendGetEvents` is enough decoupling -- I'm willing to pay the price of renaming a function in order to get pure functions.

Reply:
I've done some reflecting on this and I now believe you both are correct. The value in the gains in decoupling between the JS-SDK and the Access API as result of not having pure functions is less than the value in having pure functions. Also - the coupling in function naming to Access API method name is limited in scope to just the individual send functions, it's not such a major issue. 

I have put together a (still in progress, I want to add better error handling to it tomorrow) PR https://github.com/onflow/flow-js-sdk/pull/718 that replicates this new pattern across sendGetEvents and the other send functions that it would work with. The pattern being the splitting up of the various individual send functions for each AccessAPI method out of the exported send method. We could either point this new PRs base to this PRs branch, or merge the other PR in lieu of this PR, either works!